src/lib/client.ts

```diff
@@ -203,6 +208,8 @@ export class Client {
   }
 }
 
-export function startFarming(path: string): Promise<string> {
-  return (tauri.invoke('farming', { path })).then(value => value as string)
+export async function startFarming(path: string): Promise<ClientIdentity> {
+  const result = await (tauri.invoke('farming', { path })).then(value => value as ClientIdentity)
+  const farmer_identity: ClientIdentity = { public_key: result.public_key, mnemonic: result.mnemonic }
+  return farmer_identity
 }
```

Comment:
This should be sufficient I think:
```suggestion
export function startFarming(path: string): Promise<ClientIdentity> {
  return tauri.invoke('farming', { path }));
}
```